src/window.cpp

```diff
@@ -3180,9 +3182,8 @@ void UpdateWindows()
 	 * But still empty the invalidation queues above. */
 	if (_network_dedicated) return;
 
-	static GUITimer hundredth_timer = GUITimer(1);
-	if (hundredth_timer.Elapsed(delta_ms)) {
-		hundredth_timer.SetInterval(3000); // Historical reason: 100 * MILLISECONDS_PER_TICK
+	if (_hundredth_tick_timeout == 0) {
+		_hundredth_tick_timeout = 100;
```

Comment:
Maybe better to move this `if` inside `DecreaseWindowCounters` too? Now it is so far away from it, and I am not sure if there is a good reason for it? Well, besides the `_network_dedicated` check, that needs to be duplicated I guess?

Reply:
Something like the 3rd commit ?